app/models/miq_server/role_management.rb

```diff
@@ -239,6 +239,13 @@ def synchronize_active_roles(servers, roles_to_sync)
         elsif delta > 0
           delta.times do
             if inactive.length > 0
+              # We need to sort by:
+              #   * priority (some roles have 'affinity' to run on a specific server)
+              #   * by the number of active roles on a server
+              inactive = inactive.sort_by do |server, priority|
+                [priority, server.reload.active_roles.size]
+              end
+
```

Comment:
Thought about this a while for fun but I think any extraction does more harm than good just because of the fact that you're dealing with local vars in this context and not actual attrs on the sortables (I think).